System/Primitive.cpp

```diff
@@ -158,6 +158,102 @@ namespace RTE {
 		}
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void RoundBoxPrimitive::Draw(BITMAP *drawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			Vector drawEnd = m_EndPos - targetPos;
+
+			// Top left corner
+			arc(drawScreen, drawStart.m_X + m_CornerRadius, drawEnd.m_Y + m_CornerRadius, itofix(64), itofix(128), m_CornerRadius, m_Color);
+			// Bottom left corner
+			arc(drawScreen, drawStart.m_X + m_CornerRadius, drawStart.m_Y - m_CornerRadius, itofix(128), itofix(-64), m_CornerRadius, m_Color);
+			// Top right corner
+			arc(drawScreen, drawEnd.m_X - m_CornerRadius, drawEnd.m_Y + m_CornerRadius, itofix(0), itofix(64), m_CornerRadius, m_Color);
+			// Bottom right corner
+			arc(drawScreen, drawEnd.m_X - m_CornerRadius, drawStart.m_Y - m_CornerRadius, itofix(-64), itofix(0), m_CornerRadius, m_Color);
+
+			// Top plane
+			hline(drawScreen, drawStart.m_X + m_CornerRadius, drawStart.m_Y, drawEnd.m_X - m_CornerRadius, m_Color);	
+			// Bottom plane
+			hline(drawScreen, drawStart.m_X + m_CornerRadius, drawEnd.m_Y, drawEnd.m_X - m_CornerRadius, m_Color);
+			// Left plane
+			vline(drawScreen, drawStart.m_X, drawStart.m_Y - m_CornerRadius, drawEnd.m_Y + m_CornerRadius, m_Color);
+			// Right plane
+			vline(drawScreen, drawEnd.m_X, drawStart.m_Y - m_CornerRadius, drawEnd.m_Y + m_CornerRadius, m_Color);
+		} else {
+			Vector drawStartLeft;
+			Vector drawEndLeft;
+			Vector drawStartRight;
+			Vector drawEndRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+			TranslateCoordinates(targetPos, m_EndPos, drawEndLeft, drawEndRight);
+
+			arc(drawScreen, drawStartLeft.m_X + m_CornerRadius, drawEndLeft.m_Y + m_CornerRadius, itofix(64), itofix(128), m_CornerRadius, m_Color);
```

Comment:
If you're gonna have comments about corners and planes above, you should probably duplicate them here too. Again I'd say 1 or 2 comments tops, or none.

Reply:
No need, the ones here follow the same order as the ones above.

Reply:
Oh yeah, that's a good point. That's fair enough.